cpp/do_module/do_module.cpp

```diff
@@ -0,0 +1,391 @@
+    joined_strings_size += strings[i].size();
+  }
+
+  std::string joined_strings;
+  joined_strings.reserve(joined_strings_size + delimiter.size() * (strings.size() - 1));
+
+  for (size_t i = 1; i < strings.size(); i++) {
+    joined_strings += delimiter + strings[i];
+  }
+
+  return joined_strings;
+}
+
+std::string GetGraphFirstClassEntityAlias(const std::string &entity_name) {
+  return fmt::format("${} AS __{}_id", entity_name, entity_name);
+}
+
+std::string GetPrimitiveEntityAlias(const std::string &primitive_name) {
+  return fmt::format("${} AS {}", primitive_name, primitive_name);
+}
+
+std::string ConstructWithStatement(const ParamNames &names) {
+  std::vector<std::string> with_entity_vector;
+  for (const auto &node_name : names.node_names) {
+    with_entity_vector.emplace_back(GetGraphFirstClassEntityAlias(node_name));
+  }
+  for (const auto &rel_name : names.relationship_names) {
+    with_entity_vector.emplace_back(GetGraphFirstClassEntityAlias(rel_name));
+  }
+  for (const auto &prim_name : names.primitive_names) {
+    with_entity_vector.emplace_back(GetPrimitiveEntityAlias(prim_name));
+  }
+
+  return fmt::format("WITH {}", Join(with_entity_vector, ", "));
+}
+
+std::string ConstructMatchingNodeById(const std::string &node_name) {
+  return fmt::format("MATCH ({}) WHERE ID({}) = __{}_id", node_name, node_name, node_name);
+}
+
+std::string ConstructMatchingRelationshipById(const std::string &rel_name) {
+  return fmt::format("MATCH ()-[{}]->() WHERE ID({}) = __{}_id", rel_name, rel_name, rel_name);
+}
+
+std::string ConstructMatchGraphEntitiesById(const ParamNames &names) {
+  std::string match_string = "";
+  std::vector<std::string> match_by_id_vector;
+  for (const auto &node_name : names.node_names) {
+    match_by_id_vector.emplace_back(ConstructMatchingNodeById(node_name));
+  }
+  for (const auto &rel_name : names.relationship_names) {
+    match_by_id_vector.emplace_back(ConstructMatchingRelationshipById(rel_name));
+  }
+
+  if (match_by_id_vector.size()) {
+    match_string = Join(match_by_id_vector, " ");
+  }
+
+  return match_string;
+}
+
+std::string ConstructQueryPreffix(const ParamNames &names) {
+  if (!names.node_names.size() && !names.relationship_names.size() && !names.primitive_names.size()) {
+    return std::string();
+  }
+
+  auto with_variables = ConstructWithStatement(names);
+  auto match_string = ConstructMatchGraphEntitiesById(names);
+
+  return fmt::format("{} {}", with_variables, match_string);
+}
+
+std::string ConstructPreffixQuery(const mgp::Map &parameters) {
+  const auto param_names = ExtractParamNames(parameters);
+
+  return ConstructQueryPreffix(param_names);
+}
+
+std::string ConstructFinalQuery(const std::string &running_query, const std::string &preffix_query) {
+  return fmt::format("{} {}", preffix_query, running_query);
+}
+
+mg::Map ConstructParams(const ParamNames &param_names, const mgp::Map &parameters) {
+  mg::Map new_params{parameters.Size()};
```

Comment:
`mg::Map new_params{parameters.Size()};`
prefer initialization to doing assignment